### PR TITLE
Update BuildOverrides overview

### PR DESCRIPTION
**Pull request by mburke5678**
https://bugzilla.redhat.com/show_bug.cgi?id=1573428

### CODE REVIEW

File: install_config/build_defaults_overrides.adoc [hunk 1]
```diff
@@ -36,18 +36,38 @@ are not present on the build definition, they are set to the default value.
...
 override a setting in a build, regardless of the value stored in the build. It
-currently supports overriding the `forcePull` flag on a build strategy to
+currently supports xref:manually-setting-global-build-overrides[overriding the `forcePull` 
+flag on a build strategy] to
 enforce always refreshing the local image during a build by pulling the image
 from the registry. This ensures that a user can only build with an image that
 they are allowed to pull. The plug-in can also be configured to apply a set of
 image labels to every built image.
++
+For information on configuring he `BuildOverrides` admission control plug-in and
```
Comment (wzheng1) [inline install_config/build_defaults_overrides.adoc]: 'he' should be 'the'

**Comment by wzheng1** [comment]
Will annotations and tolerations be added in this pr? since they can also be modified in Setting Global Build Overrides.

**Comment by mburke5678** [comment]
@wzheng1 How many versions back should the changes be applied? The BuildDefaults and BuildOverrides plug-ins go back to at least 3.2

**Comment by mburke5678** [comment]
@openshift/team-documentation PTAL

File: install_config/build_defaults_overrides.adoc [hunk 2]
```diff
@@ -18,13 +18,14 @@ Developers can define settings in specific build configurations within their
...
 for Git cloning]. Rather than requiring developers to define certain settings in
-each of their build configurations, cluster administrators can use admission
+each build configuration, administrators can use admission
 control plug-ins to configure global build defaults and overrides that
 automatically use these settings in any build.
 
-The settings from these plug-ins are not set in the build configurations or
-builds themselves, but rather are only used during the build process. This
-allows administrators to change the global configuration at any time, and any
+The settings from these plug-ins are used only during the build process,
```
Comment (kalexand-rh) [inline install_config/build_defaults_overrides.adoc]: s/process,/process

File: install_config/build_defaults_overrides.adoc [hunk 3]
```diff
@@ -18,13 +18,14 @@ Developers can define settings in specific build configurations within their
...
+but are not set in the build configurations or
+builds themselves. Configuring the settings through the plug-ins
+allows administrators to change the global configuration at any time and any
```
Comment (kalexand-rh) [inline install_config/build_defaults_overrides.adoc]: s/time/time,

File: install_config/build_defaults_overrides.adoc [hunk 4]
```diff
@@ -18,13 +18,14 @@ Developers can define settings in specific build configurations within their
...
 builds that are re-run from existing build configurations or builds will get the
```
Comment (kalexand-rh) [inline install_config/build_defaults_overrides.adoc]: s/will get/are assigned

File: install_config/build_defaults_overrides.adoc [hunk 5]
```diff
@@ -35,19 +36,38 @@ values that have been configured for a specific build. However, if those values
 are not present on the build definition, they are set to the default value.
 
 * The `BuildOverrides` admission control plug-in allows administrators to
-override a setting in a build, regardless of the value stored in the build. It
-currently supports overriding the `forcePull` flag on a build strategy to
-enforce always refreshing the local image during a build by pulling the image
-from the registry. This ensures that a user can only build with an image that
+overrides a setting in a build, regardless of the value stored in the build. The plug-in
```
Comment (kalexand-rh) [inline install_config/build_defaults_overrides.adoc]: s/overrides/override

File: install_config/build_defaults_overrides.adoc [hunk 6]
```diff
@@ -35,19 +36,38 @@ values that have been configured for a specific build. However, if those values
...
+flag on a build strategy] to force refreshing the local image from the registry during a build. 
+Refreshing ensures that users can build only with images that
 they are allowed to pull. The plug-in can also be configured to apply a set of
 image labels to every built image.
++
+For information on configuring the `BuildOverrides` admission control plug-in and
+the values you can override, see
+xref:manually-setting-global-build-overrides[Manually Setting Global Build Overrides].
+
+The default node selectors and the `BuildDefaults` or `BuildOverride` admission plugins work together as follows:
```
Comment (kalexand-rh) [inline install_config/build_defaults_overrides.adoc]: s/plugins/plug-ins

File: install_config/build_defaults_overrides.adoc [hunk 7]
```diff
@@ -35,19 +36,38 @@ values that have been configured for a specific build. However, if those values
...
+
+* The default  project node selector, defined in the `projectConfig.defaultNodeSelector` field in the master configuration file,  
+is applied to the pods created in all projects without a specified `nodeSelector` value. This includes builds with `nodeSelector="null"` 
+on clusters where the `BuildDefaults` or `BuildOverride` nodeselector is not set).
```
Comment (kalexand-rh) [inline install_config/build_defaults_overrides.adoc]: s/set)/set

File: install_config/build_defaults_overrides.adoc [hunk 8]
```diff
@@ -35,19 +36,38 @@ values that have been configured for a specific build. However, if those values
...
 they are allowed to pull. The plug-in can also be configured to apply a set of
 image labels to every built image.
++
+For information on configuring the `BuildOverrides` admission control plug-in and
+the values you can override, see
+xref:manually-setting-global-build-overrides[Manually Setting Global Build Overrides].
+
+The default node selectors and the `BuildDefaults` or `BuildOverride` admission plugins work together as follows:
+
+* The default  project node selector, defined in the `projectConfig.defaultNodeSelector` field in the master configuration file,
```
Comment (kalexand-rh) [inline install_config/build_defaults_overrides.adoc]: There's an extra space between default and project.

File: install_config/build_defaults_overrides.adoc [hunk 9]
```diff
@@ -35,19 +36,38 @@ values that have been configured for a specific build. However, if those values
...
 image labels to every built image.
++
+For information on configuring the `BuildOverrides` admission control plug-in and
+the values you can override, see
+xref:manually-setting-global-build-overrides[Manually Setting Global Build Overrides].
+
+The default node selectors and the `BuildDefaults` or `BuildOverride` admission plugins work together as follows:
+
+* The default  project node selector, defined in the `projectConfig.defaultNodeSelector` field in the master configuration file,  
+is applied to the pods created in all projects without a specified `nodeSelector` value. This includes builds with `nodeSelector="null"`
```
Comment (kalexand-rh) [inline install_config/build_defaults_overrides.adoc]: "This includes builds" - I thought these settings applied to pods. Should it be "These settings are applied to builds?"

File: install_config/build_defaults_overrides.adoc [hunk 10]
```diff
@@ -35,19 +36,38 @@ values that have been configured for a specific build. However, if those values
...
+xref:manually-setting-global-build-overrides[Manually Setting Global Build Overrides].
+
+The default node selectors and the `BuildDefaults` or `BuildOverride` admission plugins work together as follows:
+
+* The default  project node selector, defined in the `projectConfig.defaultNodeSelector` field in the master configuration file,  
+is applied to the pods created in all projects without a specified `nodeSelector` value. This includes builds with `nodeSelector="null"` 
+on clusters where the `BuildDefaults` or `BuildOverride` nodeselector is not set). 
+
+* The cluster-wide default build node selector, `admissionConfig.pluginConfig.BuildDefaults.configuration.nodeSelector`, 
+is applied only if the `nodeSelector="null"` is set in the build configuration. `nodeSelector=null` is the default setting.
```
Comment (kalexand-rh) [inline install_config/build_defaults_overrides.adoc]: s/`nodeSelector="null"`/`nodeSelector="null"` parameter

File: install_config/build_defaults_overrides.adoc [hunk 11]
```diff
@@ -35,19 +36,38 @@ values that have been configured for a specific build. However, if those values
...
+
+* With a default project or cluster-wide node selector, the default is added as an AND to the build node selector,
```
Comment (kalexand-rh) [inline install_config/build_defaults_overrides.adoc]: s/default/default setting

File: install_config/build_defaults_overrides.adoc [hunk 12]
```diff
@@ -35,19 +36,38 @@ values that have been configured for a specific build. However, if those values
...
+
+* The default  project node selector, defined in the `projectConfig.defaultNodeSelector` field in the master configuration file,  
+is applied to the pods created in all projects without a specified `nodeSelector` value. This includes builds with `nodeSelector="null"` 
+on clusters where the `BuildDefaults` or `BuildOverride` nodeselector is not set). 
+
+* The cluster-wide default build node selector, `admissionConfig.pluginConfig.BuildDefaults.configuration.nodeSelector`, 
+is applied only if the `nodeSelector="null"` is set in the build configuration. `nodeSelector=null` is the default setting.
+
+* With a default project or cluster-wide node selector, the default is added as an AND to the build node selector, 
+which is set by the `BuildDefaults` or `BuildOverride` admission plugins. This means, the build will only be scheduled
```
Comment (kalexand-rh) [inline install_config/build_defaults_overrides.adoc]: s/plugins/plug-ins

Comment (kalexand-rh) [reply install_config/build_defaults_overrides.adoc]: s/ This means, the build will only be scheduled/These settings mean that the build will be scheduled only

**Review by kalexand-rh — COMMENTED** [review]
I have a couple of picks, but this looks good!

**Pull request merged**